Readme.md

```diff
@@ -1,29 +1,66 @@
+
+Click the **Parameters** button from the **Dashboard** page group on the **Ribbon**'s **Home** page. Click **Add** to create a dashboard parameter. Diasble the [Visible](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardParameter.Visible) property to hide the parameter from the **Dashboard Parameters** dialog. In this example, the dashboard parameter's name is **custIDDashboardParameter** and its default value is *ALFKI*:
+
+![Dashbaord Parameter Settings](images/dashboard-parameter-settings.png)
+
+### Create a Custom Query
+
+Custom SQL queries are disabled by default. To allow users to specify custom SQL queries in the [Query Builder](https://docs.devexpress.com/Dashboard/16152)/[Query Editor](https://docs.devexpress.com/Dashboard/115206), set the [`SqlWizardSettings.EnableCustomSql`](https://docs.devexpress.com/Dashboard/DevExpress.DataAccess.UI.Wizard.SqlWizardSettings.EnableCustomSql) property exposed by the [`DashboardDataSourceWizardSettings`](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWin.DashboardDataSourceWizardSettings) class to `true`. Add the following code to allow custom SQL queries:
+ 
+```csharp
+dashboardDesigner1.DataSourceWizard.SqlWizardSettings.EnableCustomSql = true;
+```
+
+Specify the query in the **SQL string** editor in [Query Builder](https://docs.devexpress.com/Dashboard/16152)/[Query Editor](https://docs.devexpress.com/Dashboard/115206):
+
+![Custom SQL Query with a Dashboard Parameter](images/custom-query.png)
 
+This query contains a [query parameter](https://docs.devexpress.com/Dashboard/113947) named **custIDQueryParameter**.
 
-This example shows how to filter a custom SQL query by changing a parameter value in the [DashboardViewer.CustomParameters](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWin.DashboardViewer.CustomParameters) event handler.
 
+### Bind the Query Parameter to the Dashboard Parameter
 
-The **custIDQueryParameter** query parameter is included in a **WHERE** clause of a [custom SQL query](https://docs.devexpress.com/Dashboard/115212). The **custIDQueryParameter** parameter is also bound to the hidden **custIDDashboardParameter** dashboard parameter. The value of this parameter is changed at runtime by handling the [DashboardViewer.CustomParameters](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWin.DashboardViewer.CustomParameters) event that is raised before the DashboardViewer queries a database.
+To be able to change the query parameter's value, bind it to the **custIDDashboardParameter** dashboard parameter. To do this, on the next page of the **Query Editor** click **Add** to create a query parameter. Specify the parameter [settings](https://docs.devexpress.com/Dashboard/16169/). In the **Name** column, specify the name used in the query. Select the **Expression** checkbox and bind the query parameter to the dashboard parameter:
```

Comment:
```suggestion
To be able to change the query parameter's value, bind it to the **custIDDashboardParameter** dashboard parameter. To do this, on the next page of the **Query Editor**, click **Add** to create a query parameter. Specify the parameter [settings](https://docs.devexpress.com/Dashboard/16169/). In the **Name** column, specify the name used in the query. Select the **Expression** checkbox and bind the query parameter to the dashboard parameter:
```

Reply:
1. The idea is to change the query parameter value dynamically. A user can set a static value or bind it to the dashboard parameter.

How about this:

```suggestion
To change the query parameter value dynamically, bind it to the **custIDDashboardParameter** dashboard parameter. To do this, on the next page of the **Query Editor**, click **Add** to create a query parameter. Specify the parameter [settings](https://docs.devexpress.com/Dashboard/16169/). In the **Name** column, specify the name used in the query. Select the **Expression** check box and bind the query parameter to the dashboard parameter:
```

Reply:
Sounds good. Thank you!